File: Week01/1.两数之和.cpp
/*
 * @lc app=leetcode.cn id=1 lang=cpp
 *
 * [1] 两数之和
 */

// @lc code=start
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> map;
        for(int i = 0; i < nums.size(); i++){
            if(map.count(target - nums[i])){
                return {map[target - nums[i]], i};
            }
            map[nums[i]] = i;
        }
        return {-1, -1};
    }
};
// @lc code=end

File: Week02/1.两数之和.py
#
# @lc app=leetcode.cn id=1 lang=python3
#
# [1] 两数之和
#

# @lc code=start
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        # 采用dict用map的方式实现
        numMap = {}
        for i, num in enumerate(nums):
            if numMap.get(target - num) is not None:
                return [numMap.get(target - num), i]
            numMap[num] = i
# @lc code=end

